### PR TITLE
[7274] migrate from django-ckeditor to django-ckeditor-5

### PR DESCRIPTION
**Pull request by goapunk**

### CODE REVIEW

File: adhocracy4/ckeditor/fields.py [hunk 1]
```diff
@@ -1,23 +1,10 @@
...
 
+# FIXME: remove these fields / file
 class RichTextCollapsibleMixin:
-    def __init__(self, *args, **kwargs):
-        kwargs["extra_plugins"] = kwargs.get("extra_plugins", []) + _extra_plugins
-        kwargs["external_plugin_resources"] = (
-            kwargs.get("external_plugin_resources", []) + _external_plugin_resources
-        )
-        super().__init__(*args, **kwargs)
+    pass
```
Comment (goapunk) [inline adhocracy4/ckeditor/fields.py:9]: not sure how to get rid of these fields, as they will break old migrations. One way would be to squash / reset migrations but that also has its drawbacks

Comment (goapunk) [reply adhocracy4/ckeditor/fields.py:9]: @m4ra any idea here?

Comment (m4ra) [reply adhocracy4/ckeditor/fields.py:9]: The fields that will be deleted, the data they retain they don't need to be migrated to the new CK?

Comment (goapunk) [reply adhocracy4/ckeditor/fields.py:9]: They are migrated to `CKEditor5Field`. Will look into squashing, thanks!

**Review by philli-m — APPROVED** [review]
s

**Review by philli-m — APPROVED** [review]
it's definitely working, also displays old content too but as it's quite pythony, maybe @m4ra you want to give it a final look over?

File: adhocracy4/ckeditor/storage.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+from urllib.parse import urljoin
+
+from django.conf import settings
+from django.core.files.storage import FileSystemStorage
+
+
+class CustomStorage(FileSystemStorage):
+    """Custom storage to store uploads in a subfolder called uploads"""
+
+    location = os.path.join(settings.MEDIA_ROOT, "uploads")
```
Comment (m4ra) [inline adhocracy4/ckeditor/storage.py:11]: I'd specify those are from the ck editor uploads, so maybe a prefix or a suffix to `uploads`

**Review by m4ra — COMMENTED** [review]
LGTM

File: adhocracy4/ckeditor/fields.py [hunk 3]
```diff
@@ -1,23 +1,10 @@
...
-_external_plugin_resources = [
-    (
-        "collapsibleItem",
-        "/static/ckeditor_collapsible/",
-        "plugin.js",
-    )
-]
-
 
+# FIXME: remove these fields / file
```
[the remark this answers is not in the document]
Comment (goapunk) [reply adhocracy4/ckeditor/fields.py:5]: They are empty (pass), but still there. The problem is that removing them breaks some migrations, so we need to squash them (I would do this in a separate PR when removing the old django-ckeditor package)

File: adhocracy4/ckeditor/storage.py (hunk 2 above)
Comment (goapunk) [reply adhocracy4/ckeditor/storage.py:11]: This breaks existing behavior though, the old ckeditor uploads them to `uploads`. I'd prefer to keep the current behavior for now. We have another story which will touch the upload behavior (file names, path), so we can change it as part of that it that's ok?

**Pull request merged**